go/private/rules/binary.bzl

```diff
@@ -69,10 +67,12 @@ go_binary = rule(
             cfg = "data",
         ),
         "srcs": attr.label_list(allow_files = go_filetype),
-        "deps": attr.label_list(providers = [GoLibrary]),
+        "deps": attr.label_list(providers = [GoLibrary], aspects = [go_archive_aspect]),
         "importpath": attr.string(),
-        "library": attr.label(providers = [GoLibrary]),
-        "embed": attr.label_list(providers = [GoEmbed]),
+        "library": attr.label(providers = [GoLibrary], aspects = [go_archive_aspect]),
+        "embed": attr.label_list(providers = [GoEmbed], aspects = [go_archive_aspect]),
+        "pure": attr.string(values=["on", "off", "auto"], default="auto"),
```

Comment:
It's regrettable there's not a more canonical tristate attribute type. I don't feel comfortable with strings doing this job, but I can't think of anything better. I think some of the C++ rules use integers, and I definitely prefer strings over those.

Reply:
Unfortunately, they *have* to be strings. Aspects are only allowed string type attributes, and those must be of a type that have a specified set of values...
I would rather have had mostly bool (true, false, None)
In fact, I would rather have been able to propagate the features string array, but I can't. And in fact, because only label type attributes are allowed function defaults, I can't even hide the string ones and calculate them from the features.